travis/stretch/build.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+set -e
+sudo mkhomedir_helper $(whoami)
+
+BUILD_DIR="build"
+cmake --version
+
+cd /travis
+mkdir -p ${BUILD_DIR}
+cd ${BUILD_DIR}
+cmake -DCMAKE_C_COMPILER=gcc-${CC_VER} -DCMAKE_CXX_COMPILER=g++-${CC_VER} -DCMAKE_BUILD_TYPE=Release ${CMAKE_ARGS} /travis
+make
+
+if [ -n "${TEST:+1}" ]; then
+  echo "Running Tests"
+  if [[ ${CMAKE_ARGS} == *"BUILD_PYTHON=ON"* ]]; then
+      cd /travis
+      python3 setup.py test
+  else
+      export GTEST_COLOR=1
+      ctest -VV
+  fi
+fi
+
+if [ -n "${DEPLOY:+1}" ]; then
+    cd /travis
+    python3 setup.py sdist
+fi
+
+if [ -n "${BUILD_DIST:+1}" ]; then
+    mkdir -p /travis/results # /travis is persistent
+
+    cd /travis/travis
+    tar xvf keys.tar
+    gpg --import public.gpg || true
+    gpg --import private.gpg || true
+
+    # /build is inside the Docker container. we don't want to litter the host with persistent files. Especially useful if this is your dev machine
+    # /build is already made by the Dockerfile, so let's prepare it for use by a normal user
+    sudo chown -R $(id -u):$(id -g) /build
+    cd /build
+    pip3 download --no-deps pyqrllib
```

Comment:
This is pulling pyqrllib before it has been updated. Deb packages will be always one release behind. 
The order of the stages is wrong or build_dist should be using a local package by running setup.py in the same container.